actions/setup-java@v2 - Documentation (#134)
actions/setup-java: README.md

  <a href="https://github.com/actions/setup-java"><img alt="GitHub Actions status" src="https://github.com/actions/setup-java/workflows/Main%20workflow/badge.svg"></a>
</p>

This action provides the following functionality for GitHub Actions runners:
- Downloading and setting up a requested version of Java. See [Usage](#Usage) section for the list of supported distributions

> - Extracting and caching custom version of Java from local file

Reviewer: `See [Usage](#Usage) for a list of supported distributions`. No need to have the "section" after the hyperlink

`Extracting and caching custom version of Java from a local file`

- Configuring runner for publishing using Apache Maven
- Configuring runner for publishing using Gradle
- Configuring runner for using GPG private key
- Registering problem matchers for error output

## V2 vs V1

> - V2 has support of custom distributions and provides support of Zulu OpenJDK and Adoptium (former AdoptOpenJDK) out of the box. V1 supports only Zulu OpenJDK

Reviewer: I would switch (former AdoptOpenJDK) to (formerly known as AdoptOpenJDK)

- V2 requires you to specify distribution along with the version. V1 defaults to Zulu OpenJDK, only version input is required. Follow [the migration guide](docs/switching-to-v2.md) to switch from V1 to V2

## Usage

> Input `distribution` is mandatory. See [Supported distributions](../README.md#Supported-distributions) section for the list of available options.

Reviewer: ```suggestion
Input `distribution` is mandatory. See [Supported distributions](../README.md#Supported-distributions) for a list of available options.
```

### Basic
```yaml
steps:
- uses: actions/checkout@v2
- uses: actions/setup-java@v2-preview
  with:
    distribution: 'adoptium' # See 'Supported distributions' for available options
    java-version: '11'
- run: java -cp java HelloWorldApp
```

#### Supported version syntax
Input `java-version` supports version range or exact version in [SemVer](https://semver.org/) format:
- major versions: `8`, `11`, `15`, `11.x`
- more specific versions: `8.0.232`, `11.0.4`, `11.0`, `11.0.x`

> - an early access (EA) versions: `15-ea`, `15.0.0-ea`, `15.0.0-ea.2`

Reviewer: `The "java-verion" input supports an exact version or a version range using [SemVer](https://semver.org/) notation:`

the last line I would change to just `early access...`
